File: 백준/Silver/18258. 큐 2/큐 2.py
import sys
from collections import deque

n = int(sys.stdin.readline())
queue = deque()

for _ in range(n):
    command = sys.stdin.readline().split()

    if command[0] == "push":
        queue.append(command[1])
    elif command[0] == "pop":
        if len(queue) == 0:
            sys.stdout.write("-1\n")
        else:
            sys.stdout.write(f"{queue.popleft()}\n")
    elif command[0] == "size":
        sys.stdout.write(f"{len(queue)}\n")
    elif command[0] == "empty":
        if len(queue) == 0:
            sys.stdout.write("1\n")
        else:
            sys.stdout.write("0\n")
    elif command[0] == "front":
        if len(queue) == 0:
            sys.stdout.write("-1\n")
        else:
            sys.stdout.write(f"{queue[0]}\n")
    elif command[0] == "back":
        if len(queue) == 0:
            sys.stdout.write("-1\n")
        else:
            sys.stdout.write(f"{queue[-1]}\n")

File: 백준/Silver/18258. 큐 2/큐 2.cc
#include <iostream>
#include <queue>
using namespace std;

int main() {
    ios::sync_with_stdio(false); cin.tie(NULL); cout.tie(NULL);
    
    int n;
    queue<int> q;
    cin >> n;

    for (int i = 0; i < n; i++) {
        string command;
        cin >> command;

        if (command == "push") {
            int num;
            cin >> num;
            q.push(num);
        }
        else if (command == "pop") {
            if (q.size() == 0)
                cout << -1 << "\n";
            else {
                cout << q.front() << "\n";
                q.pop();
            }
        }
        else if (command == "size") {
            cout << q.size() << "\n";
        }
        else if (command == "empty") {
            cout << q.empty() << "\n";
        }
        else if (command == "front") {
            if (q.empty())
                cout << -1 << "\n";
            else
                cout << q.front() << "\n";
        }
        else if (command == "back") {
            if (q.empty())
                cout << -1 << "\n";
            else
                cout << q.back() << "\n";
        }
    }
    
    return 0;
}